### PR TITLE
Throw error if custom port cannot be used

### LINKED ISSUE #5495: Devserver should not find a new port if a specific one is requested
# 🐛 bug report

Parcel's devserver finds another suitable port if one is not available. This makes sense in the default case when a specific port number is not requested.

## 🎛 Configuration (.babelrc, package.json, cli command)

To reproduce, run

`parcel serve src/index.html --port 3000`

and then run it again: `parcel serve src/index.html --port 3000`

## 🤔 Expected Behavior

We should consider exiting the process with a non-zero status code. This "fail fast" signal is useful if the Parcel devserver is being expected to be listening on a specific port, rather than going ahead with starting Parcel. This might occur if the devserver is behind a reverse proxy, running in a container with a specific published port, or if running the devserver is automated, such as if it's run by a supervisor or init process.

## 😯 Current Behavior

The devserver finds another suitable port and listens on it.

## 💁 Possible Solution

Exit the process with a non-zero code and do not bind on another port.

### PR DESCRIPTION
**Pull request by DeMoorJasper**
# ↪️ Pull Request

This PR is based on the work in #5516 to fix a "bug" where parcel does not exit if the custom user-defined port cannot be used.

Closes #5495

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 4.14s | +133.00ms |
| Cached | 788.00ms | +3.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 64.00ms | +25.00ms ⚠️ |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 50.00ms | -28.00ms 🚀 |
| dist/legacy/index.7e103b24.js | 1.07kb | +0.00b | 1.37s | +87.00ms ⚠️ |
| dist/modern/index.97edf045.js | 1.07kb | +0.00b | 1.32s | -120.00ms 🚀 |
| dist/legacy/index.html | 701.00b | +0.00b | 1.37s | +82.00ms ⚠️ |
| dist/modern/index.html | 701.00b | +0.00b | 1.32s | -120.00ms 🚀 |
| dist/legacy/index.507458dd.css | 77.00b | +0.00b | 1.37s | +82.00ms ⚠️ |
| dist/modern/index.8ca944bc.css | 77.00b | +0.00b | 1.32s | -120.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 51.00ms | -5.00ms 🚀 |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 51.00ms | -5.00ms 🚀 |
| dist/legacy/index.7e103b24.js | 1.07kb | +0.00b | 75.00ms | -11.00ms 🚀 |
| dist/modern/index.97edf045.js | 1.07kb | +0.00b | 75.00ms | -4.00ms 🚀 |
| dist/legacy/index.html | 701.00b | +0.00b | 76.00ms | -9.00ms 🚀 |
| dist/modern/index.html | 701.00b | +0.00b | 75.00ms | -4.00ms 🚀 |
| dist/legacy/index.507458dd.css | 77.00b | +0.00b | 75.00ms | -10.00ms 🚀 |
| dist/modern/index.8ca944bc.css | 77.00b | +0.00b | 75.00ms | -4.00ms 🚀 |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 22.72s | +339.00ms |
| Cached | 12.13s | -8.66s 🚀 |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/NotFound.5d9b792b.js | 532.00b | +0.00b | 323.00ms | +19.00ms ⚠️ |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 67.00ms | +16.00ms ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.js | 479.93kb | +0.00b | 65.00ms | -8.56s 🚀 |
| dist/PermalinkedComment.62e59ad1.js | 4.22kb | +0.00b | 65.00ms | -30.00ms 🚀 |
| dist/UserProfile.7223b396.js | 1.70kb | +0.00b | 64.00ms | -31.00ms 🚀 |
| dist/NotFound.5d9b792b.js | 532.00b | +0.00b | 38.00ms | -5.00ms 🚀 |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 47.00ms | +3.00ms ⚠️ |

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.13m | -298.00ms |
| Cached | 3.08s | +168.00ms ⚠️ |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/workerHasher.8a9cd633.js | 11.90kb | +0.00b | 7.12s | -737.00ms 🚀 |
| dist/card.f50bfb7e.js | 5.71kb | +0.00b | 11.52s | -594.00ms 🚀 |
| dist/simpleHasher.4e275b4f.js | 755.00b | +0.00b | 7.11s | -739.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.e924e0e1.js | 2.31mb | +0.00b | 560.00ms | +66.00ms ⚠️ |
| dist/pdfRenderer.50b8173f.js | 1.11mb | +0.00b | 632.00ms | +46.00ms ⚠️ |
| dist/editorView.267a5df1.js | 587.39kb | +0.00b | 654.00ms | +42.00ms ⚠️ |
| dist/popup.26230ab5.js | 169.21kb | +0.00b | 668.00ms | +65.00ms ⚠️ |
| dist/EmojiPickerComponent.4faa5472.js | 139.08kb | +0.00b | 619.00ms | +57.00ms ⚠️ |
| dist/Toolbar.cd51d763.js | 99.13kb | +0.00b | 654.00ms | +42.00ms ⚠️ |
| dist/media-viewer.0aa3923d.js | 72.60kb | +0.00b | 631.00ms | +45.00ms ⚠️ |
| dist/card.33239941.js | 53.89kb | +0.00b | 620.00ms | +35.00ms ⚠️ |
| dist/card.6faf41b0.js | 51.65kb | +0.00b | 643.00ms | +40.00ms ⚠️ |
| dist/Modal.f72d1800.js | 38.79kb | +0.00b | 563.00ms | +58.00ms ⚠️ |
| dist/component.effe7aa0.js | 30.87kb | +0.00b | 563.00ms | +70.00ms ⚠️ |
| dist/esm.d4fe111d.js | 27.71kb | +0.00b | 619.00ms | +57.00ms ⚠️ |
| dist/component.7d45f63d.js | 22.40kb | +0.00b | 502.00ms | +41.00ms ⚠️ |
| dist/DatePicker.7146e115.js | 21.03kb | +0.00b | 619.00ms | +58.00ms ⚠️ |
| dist/smartMediaEditor.dafbb08e.js | 16.69kb | +0.00b | 654.00ms | +42.00ms ⚠️ |
| dist/js.aaf5ecd1.js | 16.55kb | +0.00b | 502.00ms | +41.00ms ⚠️ |
| dist/dropzone.585b081a.js | 15.96kb | +0.00b | 668.00ms | +66.00ms ⚠️ |
| dist/ui.9353c8f7.js | 14.11kb | +0.00b | 620.00ms | +38.00ms ⚠️ |
| dist/workerHasher.8a9cd633.js | 11.90kb | +0.00b | 612.00ms | +51.00ms ⚠️ |
| dist/component.adc3ddbc.js | 6.27kb | +0.00b | 503.00ms | +41.00ms ⚠️ |
| dist/card.f50bfb7e.js | 5.71kb | +0.00b | 620.00ms | +34.00ms ⚠️ |
| dist/media-viewer.b1642a16.js | 4.01kb | +0.00b | 632.00ms | +35.00ms ⚠️ |
| dist/EmojiPickerComponent.acb2e5c6.js | 3.56kb | +0.00b | 613.00ms | +31.00ms ⚠️ |
| dist/png-chunks-extract.1a612bb6.js | 3.55kb | +0.00b | 502.00ms | +41.00ms ⚠️ |
| dist/index.c1492881.css | 3.46kb | +0.00b | 662.00ms | +45.00ms ⚠️ |
| dist/dropzone.ab62bdb4.js | 3.44kb | +0.00b | 668.00ms | +58.00ms ⚠️ |
| dist/Modal.99a80838.js | 3.15kb | +0.00b | 494.00ms | +48.00ms ⚠️ |
| dist/clipboard.d10471c4.js | 2.97kb | +0.00b | 668.00ms | +66.00ms ⚠️ |
| dist/16.0cf912f4.js | 2.49kb | +0.00b | 495.00ms | +49.00ms ⚠️ |
| dist/ResourcedEmojiComponent.e6000ed9.js | 2.15kb | +0.00b | 613.00ms | +31.00ms ⚠️ |
| dist/date.90ec1302.js | 1.96kb | +0.00b | 573.00ms | +60.00ms ⚠️ |
| dist/images.c69c067f.js | 1.90kb | +0.00b | 586.00ms | +66.00ms ⚠️ |
| dist/feedback.a4a4dbc3.js | 1.86kb | +0.00b | 612.00ms | +51.00ms ⚠️ |
| dist/16.2daaa966.js | 1.86kb | +0.00b | 557.00ms | +68.00ms ⚠️ |
| dist/browser.d4b2ac0a.js | 1.82kb | +0.00b | 643.00ms | +40.00ms ⚠️ |
| dist/16.72628d37.js | 1.79kb | +0.00b | 536.00ms | +60.00ms ⚠️ |
| dist/workerHasher.6aa37edc.js | 1.75kb | +0.00b | 619.00ms | +58.00ms ⚠️ |
| dist/list-number.d6f09d2f.js | 1.68kb | +0.00b | 586.00ms | +43.00ms ⚠️ |
| dist/status.09dffefc.js | 1.68kb | +0.00b | 602.00ms | +74.00ms ⚠️ |
| dist/code.b1fce661.js | 1.61kb | +0.00b | 573.00ms | +60.00ms ⚠️ |
| dist/link.18257706.js | 1.53kb | +0.00b | 587.00ms | +71.00ms ⚠️ |
| dist/heading6.ee7c597d.js | 1.53kb | +0.00b | 601.00ms | +40.00ms ⚠️ |
| dist/heading3.5a833b7b.js | 1.51kb | +0.00b | 600.00ms | +57.00ms ⚠️ |
| dist/16.542d3ab7.js | 1.51kb | +0.00b | 557.00ms | +69.00ms ⚠️ |
| dist/16.f683480d.js | 1.46kb | +0.00b | 536.00ms | +60.00ms ⚠️ |
| dist/16.72d89b1f.js | 1.46kb | +0.00b | 536.00ms | +59.00ms ⚠️ |
| dist/emoji.a556f6a6.js | 1.45kb | +0.00b | 586.00ms | +66.00ms ⚠️ |
| dist/16.35c1a785.js | 1.45kb | +0.00b | 494.00ms | +48.00ms ⚠️ |
| dist/16.e7cb0586.js | 1.44kb | +0.00b | 573.00ms | +75.00ms ⚠️ |
| dist/16.388091fb.js | 1.40kb | +0.00b | 573.00ms | +75.00ms ⚠️ |
| dist/heading5.368d12b0.js | 1.40kb | +0.00b | 601.00ms | +40.00ms ⚠️ |
| dist/expand.abd57bf3.js | 1.38kb | +0.00b | 619.00ms | +58.00ms ⚠️ |
| dist/16.493f9d2a.js | 1.36kb | +0.00b | 572.00ms | +74.00ms ⚠️ |
| dist/16.1e8baed6.js | 1.34kb | +0.00b | 494.00ms | +48.00ms ⚠️ |
| dist/heading2.de835502.js | 1.33kb | +0.00b | 600.00ms | +57.00ms ⚠️ |
| dist/16.0c3e9d9e.js | 1.32kb | +0.00b | 573.00ms | +75.00ms ⚠️ |
| dist/16.56cee737.js | 1.31kb | +0.00b | 573.00ms | +60.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.2fc3699c.js | 1.30kb | +0.00b | 620.00ms | +34.00ms ⚠️ |
| dist/mention.0381ad00.js | 1.29kb | +0.00b | 594.00ms | +50.00ms ⚠️ |
| dist/heading4.abd6ed5d.js | 1.29kb | +0.00b | 601.00ms | +58.00ms ⚠️ |
| dist/Modal.c5a75f57.js | 1.28kb | +0.00b | 502.00ms | +41.00ms ⚠️ |
| dist/16.a785c6a9.js | 1.27kb | +0.00b | 494.00ms | +48.00ms ⚠️ |
| dist/layout.6fc445d4.js | 1.27kb | +0.00b | 586.00ms | +65.00ms ⚠️ |
| dist/16.350f9e86.js | 1.27kb | +0.00b | 557.00ms | +68.00ms ⚠️ |
| dist/16.d2bd05f5.js | 1.26kb | +0.00b | 536.00ms | +59.00ms ⚠️ |
| dist/16.0c59612d.js | 1.26kb | +0.00b | 557.00ms | +69.00ms ⚠️ |
| dist/16.0a326f87.js | 1.26kb | +0.00b | 557.00ms | +68.00ms ⚠️ |
| dist/16.540cfafc.js | 1.26kb | +0.00b | 535.00ms | +59.00ms ⚠️ |
| dist/divider.2e530f7c.js | 1.25kb | +0.00b | 586.00ms | +66.00ms ⚠️ |
| dist/quote.df9cfef0.js | 1.25kb | +0.00b | 602.00ms | +74.00ms ⚠️ |
| dist/component.9c6ba2f0.js | 1.23kb | +0.00b | 560.00ms | +67.00ms ⚠️ |
| dist/action.b6fa0d15.js | 1.23kb | +0.00b | 573.00ms | +60.00ms ⚠️ |
| dist/16.76d22cc2.js | 1.23kb | +0.00b | 560.00ms | +67.00ms ⚠️ |
| dist/decision.0dfb05c0.js | 1.21kb | +0.00b | 573.00ms | +53.00ms ⚠️ |
| dist/panel-warning.d1deb7d8.js | 1.21kb | +0.00b | 595.00ms | +68.00ms ⚠️ |
| dist/16.b4718b96.js | 1.18kb | +0.00b | 573.00ms | +75.00ms ⚠️ |
| dist/list.fc45edf6.js | 1.18kb | +0.00b | 587.00ms | +44.00ms ⚠️ |
| dist/heading1.cc39194f.js | 1.18kb | +0.00b | 595.00ms | +52.00ms ⚠️ |
| dist/panel-error.d22bf8a5.js | 1.11kb | +0.00b | 595.00ms | +50.00ms ⚠️ |
| dist/panel.b435337e.js | 1.10kb | +0.00b | 601.00ms | +73.00ms ⚠️ |
| dist/table.964f293b.js | 1.09kb | +0.00b | 602.00ms | +59.00ms ⚠️ |
| dist/media-picker-analytics-error-boundary.133a9728.js | 1.05kb | +0.00b | 668.00ms | +66.00ms ⚠️ |
| dist/panel-success.0b507c40.js | 1.05kb | +0.00b | 595.00ms | +68.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.64bc429c.js | 1.05kb | +0.00b | 620.00ms | +34.00ms ⚠️ |
| dist/panel-note.07086709.js | 1.05kb | +0.00b | 595.00ms | +45.00ms ⚠️ |
| dist/simpleHasher.4e275b4f.js | 755.00b | +0.00b | 613.00ms | +44.00ms ⚠️ |
| dist/index.html | 119.00b | +0.00b | 492.00ms | +70.00ms ⚠️ |

</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 15.27s | +177.00ms |
| Cached | 698.00ms | -31.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/Three.js | 580.68kb | +0.00b | 107.00ms | -7.00ms 🚀 |

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/285788764974875149)**

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 4.16s | -22.00ms |
| Cached | 778.00ms | +1.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 70.00ms | +16.00ms ⚠️ |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 51.00ms | -27.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 53.00ms | -16.00ms 🚀 |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 79.00ms | +31.00ms ⚠️ |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 23.05s | -64.00ms |
| Cached | 14.13s | -7.48s 🚀 |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/NotFound.5d9b792b.js | 532.00b | +0.00b | 272.00ms | -35.00ms 🚀 |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 60.00ms | -22.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.js | 479.93kb | +0.00b | 82.00ms | -8.91s 🚀 |
| dist/PermalinkedComment.62e59ad1.js | 4.22kb | +0.00b | 81.00ms | -17.00ms 🚀 |
| dist/UserProfile.7223b396.js | 1.70kb | +0.00b | 81.00ms | -17.00ms 🚀 |
| dist/NotFound.5d9b792b.js | 532.00b | +0.00b | 36.00ms | -32.00ms 🚀 |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 60.00ms | +27.00ms ⚠️ |

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 2.14m | -2.43s |
| Cached | 2.97s | +41.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.d610afbe.js | 2.34mb | -4.00b 🚀 | 1.30m | -85.00ms |
| dist/media-viewer.a564b877.js | 72.71kb | +0.00b | 1.30m | +1.07m ⚠️ |
| dist/card.ac091eac.js | 53.76kb | +0.00b | 11.96s | -1.10m 🚀 |
| dist/DatePicker.27ec5589.js | 21.16kb | +0.00b | 8.67s | -1.86s 🚀 |
| dist/dropzone.27b639af.js | 16.03kb | +0.00b | 45.05s | +19.50s ⚠️ |
| dist/ui.9353c8f7.js | 14.11kb | +0.00b | 11.95s | -1.10m 🚀 |
| dist/workerHasher.8a9cd633.js | 11.90kb | +0.00b | 8.86s | -504.00ms 🚀 |
| dist/component.adc3ddbc.js | 6.27kb | +0.00b | 5.63s | +909.00ms ⚠️ |
| dist/card.f50bfb7e.js | 5.71kb | +0.00b | 1.30m | +1.07m ⚠️ |
| dist/media-viewer.b1642a16.js | 4.01kb | +0.00b | 12.54s | -1.42s 🚀 |
| dist/EmojiPickerComponent.f4a014bd.js | 3.68kb | +0.00b | 8.92s | -539.00ms 🚀 |
| dist/dropzone.ab62bdb4.js | 3.44kb | +0.00b | 16.74s | +887.00ms ⚠️ |
| dist/ResourcedEmojiComponent.e6000ed9.js | 2.15kb | +0.00b | 11.95s | -1.10m 🚀 |
| dist/card.ac215339.js | 2.15kb | +0.00b | 14.59s | -1.09s 🚀 |
| dist/date.90ec1302.js | 1.96kb | +0.00b | 5.58s | +358.00ms ⚠️ |
| dist/feedback.a4a4dbc3.js | 1.86kb | +0.00b | 8.14s | -2.37s 🚀 |
| dist/16.2daaa966.js | 1.86kb | +0.00b | 5.30s | +353.00ms ⚠️ |
| dist/16.72628d37.js | 1.79kb | +0.00b | 5.67s | +911.00ms ⚠️ |
| dist/workerHasher.6aa37edc.js | 1.75kb | +0.00b | 8.85s | -468.00ms 🚀 |
| dist/list-number.d6f09d2f.js | 1.68kb | +0.00b | 5.71s | +379.00ms ⚠️ |
| dist/link.18257706.js | 1.53kb | +0.00b | 5.71s | +385.00ms ⚠️ |
| dist/16.542d3ab7.js | 1.51kb | +0.00b | 5.14s | -1.03s 🚀 |
| dist/16.f683480d.js | 1.46kb | +0.00b | 5.63s | +909.00ms ⚠️ |
| dist/16.72d89b1f.js | 1.46kb | +0.00b | 5.14s | -929.00ms 🚀 |
| dist/emoji.a556f6a6.js | 1.45kb | +0.00b | 5.58s | +360.00ms ⚠️ |
| dist/16.e7cb0586.js | 1.44kb | +0.00b | 5.47s | +376.00ms ⚠️ |
| dist/16.388091fb.js | 1.40kb | +0.00b | 5.47s | +376.00ms ⚠️ |
| dist/expand.abd57bf3.js | 1.38kb | +0.00b | 8.14s | -2.40s 🚀 |
| dist/16.493f9d2a.js | 1.36kb | +0.00b | 5.30s | +353.00ms ⚠️ |
| dist/16.0c3e9d9e.js | 1.32kb | +0.00b | 5.30s | +357.00ms ⚠️ |
| dist/16.56cee737.js | 1.31kb | +0.00b | 5.47s | +375.00ms ⚠️ |
| dist/media-card-analytics-error-boundary.2fc3699c.js | 1.30kb | +0.00b | 1.30m | +1.07m ⚠️ |
| dist/mention.0381ad00.js | 1.29kb | +0.00b | 5.72s | +298.00ms ⚠️ |
| dist/layout.6fc445d4.js | 1.27kb | +0.00b | 5.71s | +385.00ms ⚠️ |
| dist/16.350f9e86.js | 1.27kb | +0.00b | 5.30s | +353.00ms ⚠️ |
| dist/16.d2bd05f5.js | 1.26kb | +0.00b | 5.14s | -929.00ms 🚀 |
| dist/16.0c59612d.js | 1.26kb | +0.00b | 5.14s | -929.00ms 🚀 |
| dist/16.540cfafc.js | 1.26kb | +0.00b | 5.63s | +909.00ms ⚠️ |
| dist/divider.2e530f7c.js | 1.25kb | +0.00b | 5.58s | +359.00ms ⚠️ |
| dist/action.b6fa0d15.js | 1.23kb | +0.00b | 5.48s | +377.00ms ⚠️ |
| dist/decision.0dfb05c0.js | 1.21kb | +0.00b | 5.58s | +358.00ms ⚠️ |
| dist/panel-warning.d1deb7d8.js | 1.21kb | +0.00b | 6.86s | +1.37s ⚠️ |
| dist/list.fc45edf6.js | 1.18kb | +0.00b | 5.72s | +387.00ms ⚠️ |
| dist/panel-error.d22bf8a5.js | 1.11kb | +0.00b | 6.82s | +1.39s ⚠️ |
| dist/media-picker-analytics-error-boundary.133a9728.js | 1.05kb | +0.00b | 45.05s | +19.50s ⚠️ |
| dist/panel-success.0b507c40.js | 1.05kb | +0.00b | 6.82s | +1.39s ⚠️ |
| dist/panel-note.07086709.js | 1.05kb | +0.00b | 6.82s | +1.39s ⚠️ |
| dist/media-viewer-analytics-error-boundary.c9f83b12.js | 995.00b | +0.00b | 14.59s | -1.09s 🚀 |
| dist/simpleHasher.4e275b4f.js | 755.00b | +0.00b | 8.86s | -466.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.c88da612.js | 2.31mb | +0.00b | 539.00ms | +65.00ms ⚠️ |
| dist/Modal.f72d1800.js | 38.79kb | +0.00b | 441.00ms | -42.00ms 🚀 |
| dist/component.effe7aa0.js | 30.87kb | +0.00b | 540.00ms | +63.00ms ⚠️ |
| dist/component.7d45f63d.js | 22.40kb | +0.00b | 442.00ms | -41.00ms 🚀 |
| dist/js.aaf5ecd1.js | 16.55kb | +0.00b | 442.00ms | -40.00ms 🚀 |
| dist/component.adc3ddbc.js | 6.27kb | +0.00b | 543.00ms | +60.00ms ⚠️ |
| dist/png-chunks-extract.1a612bb6.js | 3.55kb | +0.00b | 442.00ms | -41.00ms 🚀 |
| dist/Modal.99a80838.js | 3.15kb | +0.00b | 432.00ms | -34.00ms 🚀 |
| dist/16.0cf912f4.js | 2.49kb | +0.00b | 433.00ms | -33.00ms 🚀 |
| dist/16.2daaa966.js | 1.86kb | +0.00b | 468.00ms | -32.00ms 🚀 |
| dist/16.72628d37.js | 1.79kb | +0.00b | 457.00ms | -28.00ms 🚀 |
| dist/status.09dffefc.js | 1.68kb | +0.00b | 518.00ms | -35.00ms 🚀 |
| dist/link.18257706.js | 1.53kb | +0.00b | 502.00ms | -27.00ms 🚀 |
| dist/16.542d3ab7.js | 1.51kb | +0.00b | 467.00ms | -33.00ms 🚀 |
| dist/16.f683480d.js | 1.46kb | +0.00b | 457.00ms | -28.00ms 🚀 |
| dist/16.72d89b1f.js | 1.46kb | +0.00b | 457.00ms | -43.00ms 🚀 |
| dist/16.35c1a785.js | 1.45kb | +0.00b | 433.00ms | -33.00ms 🚀 |
| dist/16.e7cb0586.js | 1.44kb | +0.00b | 477.00ms | -36.00ms 🚀 |
| dist/16.388091fb.js | 1.40kb | +0.00b | 477.00ms | -36.00ms 🚀 |
| dist/16.493f9d2a.js | 1.36kb | +0.00b | 477.00ms | -26.00ms 🚀 |
| dist/16.1e8baed6.js | 1.34kb | +0.00b | 433.00ms | -33.00ms 🚀 |
| dist/16.0c3e9d9e.js | 1.32kb | +0.00b | 477.00ms | -26.00ms 🚀 |
| dist/Modal.c5a75f57.js | 1.28kb | +0.00b | 442.00ms | -40.00ms 🚀 |
| dist/16.a785c6a9.js | 1.27kb | +0.00b | 433.00ms | -33.00ms 🚀 |
| dist/layout.6fc445d4.js | 1.27kb | +0.00b | 502.00ms | -27.00ms 🚀 |
| dist/16.350f9e86.js | 1.27kb | +0.00b | 468.00ms | -32.00ms 🚀 |
| dist/16.d2bd05f5.js | 1.26kb | +0.00b | 457.00ms | -43.00ms 🚀 |
| dist/16.0c59612d.js | 1.26kb | +0.00b | 467.00ms | -33.00ms 🚀 |
| dist/16.0a326f87.js | 1.26kb | +0.00b | 467.00ms | -33.00ms 🚀 |
| dist/16.540cfafc.js | 1.26kb | +0.00b | 457.00ms | -28.00ms 🚀 |
| dist/quote.df9cfef0.js | 1.25kb | +0.00b | 518.00ms | -35.00ms 🚀 |
| dist/component.9c6ba2f0.js | 1.23kb | +0.00b | 539.00ms | +56.00ms ⚠️ |
| dist/16.76d22cc2.js | 1.23kb | +0.00b | 541.00ms | +67.00ms ⚠️ |
| dist/16.b4718b96.js | 1.18kb | +0.00b | 477.00ms | -26.00ms 🚀 |
| dist/index.html | 119.00b | +0.00b | 437.00ms | +35.00ms ⚠️ |

</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 15.45s | -67.00ms |
| Cached | 689.00ms | -49.00ms 🚀 |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/Three.js | 580.68kb | +0.00b | 95.00ms | -11.00ms 🚀 |

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/285788768117457421)**

File: packages/core/parcel/src/cli.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ var hmrOptions = {
   '-p, --port <port>': [
     'set the port to serve on. defaults to $PORT or 1234',
     value => parseInt(value, 10),
-    parseInt(process.env.PORT, 10) || 1234,
+    process.env.PORT != null ? parseInt(process.env.PORT, 10) : undefined,
```
Comment (mischnic) [inline packages/core/parcel/src/cli.js:101]: What should happen for `PORT=invalid parcel ....`? Previously, it used `NaN || 1234 === 1234`.

Comment (DeMoorJasper) [reply packages/core/parcel/src/cli.js:101]: I think we should just error and exit out in that case, which it probably already does, but I made it a bit more explicit

File: packages/core/parcel/src/cli.js [hunk 2]
```diff
@@ -343,15 +343,27 @@ async function normalizeOptions(command): Promise<InitialParcelOptions> {
   }
 
   let serve = false;
-  let {port, host} = command;
+  let {port = 1234, host} = command;
+
+  // Throw an error if port value is invalid...
+  if (!Number.isInteger(port)) {
+    throw new Error(`Defined port is not a valid integer`);
```
Comment (mischnic) [inline packages/core/parcel/src/cli.js:350]:
```suggestion
    throw new Error(`Defined port '${port}' is not a valid integer`);
```

Comment (DeMoorJasper) [reply packages/core/parcel/src/cli.js:350]: yeah tried that but we parse it so it would always return `NaN`

Comment (mischnic) [reply packages/core/parcel/src/cli.js:350]: We could do this:
```diff
diff --git packages/core/parcel/src/cli.js packages/core/parcel/src/cli.js
index b65e2e1d2..bd9a7b3f4 100755
--- packages/core/parcel/src/cli.js
+++ packages/core/parcel/src/cli.js
@@ -93,12 +93,20 @@ const commonOptions = {
   ],
 };
 
+function parse(s) {
+  let v = Number(s);
+  if (!Number.isInteger(v)) {
+    throw new Error(`Defined port ${s} is not a valid integer`);
+  }
+  return v;
+}
+
 var hmrOptions = {
   '--no-hmr': 'disable hot module replacement',
   '-p, --port <port>': [
     'set the port to serve on. defaults to $PORT or 1234',
-    value => parseInt(value, 10),
-    process.env.PORT != null ? parseInt(process.env.PORT, 10) : undefined,
+    value => parse(value),
+    process.env.PORT != null ? parse(process.env.PORT) : undefined,
   ],
   '--host <host>':
     'set the host to listen on, defaults to listening on all interfaces',
@@ -345,11 +353,6 @@ async function normalizeOptions(command): Promise<InitialParcelOptions> {
   let serve = false;
   let {port = 1234, host} = command;
 
-  // Throw an error if port value is invalid...
-  if (!Number.isInteger(port)) {
-    throw new Error(`Defined port is not a valid integer`);
-  }
-
   // Ensure port is usable
   let originalPort = port;
   if (command.name() === 'serve' || command.hmr) {
```

Comment (DeMoorJasper) [reply packages/core/parcel/src/cli.js:350]: Sure, will refactor it

Comment (DeMoorJasper) [reply packages/core/parcel/src/cli.js]: Changed it, moved parsing into normaliseOptions because of not wanting uncaught exceptions

**Review by mischnic — APPROVED** [review]

**Pull request merged**